server/src/models/employer_timeline_entry.model.js

```diff
@@ -0,0 +1,106 @@
+require("dotenv").config();
+const { DataTypes } = require("sequelize");
+const { sequelize } = require("../configs/sequelize");
+const User = require("./user.model");
+const Client = require("./client.model");
+const JobLead = require("./job_lead.model");
+const Employer = require("./employer.model");
+
+const EmployerTimelineEntry = sequelize.define(
+  "employer_timeline_entries",
+  {
+    id: {
+      type: DataTypes.BIGINT,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    date_added: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    type: {
+      type: DataTypes.ENUM("contact", "job_lead_add", "placement", "note"),
+      allowNull: false,
+    },
+    title: {
+      type: DataTypes.STRING(255),
+      allowNull: false,
+    },
+    body: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    contact: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      references: {
+        model: User, // to be replaced when Contact api is done
+        key: "id",
+      },
+      defaultValue: -1,
+    },
+    employer: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      references: {
+        model: Employer,
+        key: "id",
+      },
+      defaultValue: -1,
+    },
+    client: {
+      type: DataTypes.INTEGER,
+      allowNull: false,
+      references: {
+        model: Client,
+        key: "id",
+      },
+      defaultValue: -1,
+      validate: {
```

Comment:
We probably shouldn't throw an error here since it is pretty common to not specify a client (for example when creating a note). The same is the case for job_lead or in any other timeline model where one of the associated objects isn't always passed in.